src/nvim/api/private/helpers.c

```diff
@@ -615,14 +615,24 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
     case kObjectTypeBuffer:
     case kObjectTypeWindow:
     case kObjectTypeTabpage:
+      if (obj.data.integer > INT_MAX || obj.data.integer < 0) {
```

Comment:
I don't think checking that handles are nonnegative warrants having a separate switch arm. If checked, this should be checked at Object construction, here it is too late. Conversion should check that it fits the target domain, that is VARNUMBER_MIN` to `VARNUMBER_MAX`.